src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt

```diff
@@ -116,6 +120,54 @@ open class IntelliJPlugin : Plugin<Project> {
         }
     }
 
+    private fun checkKotlinVersion(project: Project) {
+        project.afterEvaluate {
+            // Current project is Kotlin-based, as Kotlin JVM plugin is loaded
+            project.pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+                val setupDependenciesTaskProvider =
+                    project.tasks.named<SetupDependenciesTask>(IntelliJPluginConstants.SETUP_DEPENDENCIES_TASK_NAME)
+                val setupDependenciesTask = setupDependenciesTaskProvider.get()
+                val kotlinCompileTaskProvider = project.tasks.named<KotlinCompile>("compileKotlin")
+                val kotlinCompileTask = kotlinCompileTaskProvider.get()
+
+                val ideVersion = setupDependenciesTask.idea.get().version
+                val ideKotlinVersion = getBundledKotlinStdlib(ideVersion)?.let(Version.Companion::parse)
+                val ideKotlinVersionNoPatch = ideKotlinVersion?.run { Version(major, minor, 0) } ?: Version()
+                val kotlinVersion = project.getKotlinPluginVersion().let(Version.Companion::parse)
+                val kotlinVersionNoPatch = Version(kotlinVersion.major, kotlinVersion.minor, 0)
+                val kotlinApiVersion = kotlinCompileTask.kotlinOptions.apiVersion?.let(Version.Companion::parse)
+                val kotlinLanguageVersion = kotlinCompileTask.kotlinOptions.languageVersion?.let(Version.Companion::parse)
+                val kotlinBundled = project.property("kotlin.stdlib.default.dependency").toString().toBoolean()
+
+                println("kotlinLanguageVersion='${kotlinLanguageVersion}'")
+                println("kotlinApiVersion='${kotlinApiVersion}'")
+
+                val stdlib = "Kotlin Standard Library (kotlin-stdlib)"
+                val ide = "IntelliJ Platform ($ideVersion)"
+
+                when (kotlinBundled) {
+                    true -> when (ideKotlinVersion) {
+                        null -> null // This is fine. Kotlin not bundled in the given version as: Since Kotlin 1.4, a dependency on the standard library stdlib is added automatically.
+                        else -> "You bundle $stdlib with your plugin. Please consider removing it to avoid potential issues in the runtime and decrease the size of your archive."
+                    }
+
+                    false -> when {
+                        ideKotlinVersion == null -> "$stdlib is not bundled in the $ide."
```

Comment:
I assume we cannot access exact configuration of `intellij` here? so we could print out "target platform XYZ in version Z.Y." or similar?